CMakeLists.txt

```diff
@@ -0,0 +1,167 @@
+cmake_minimum_required(VERSION 3.17)
+
+include(FetchContent)
+include(ExternalProject)
+
+# Detect the system architecture
+EXECUTE_PROCESS(
+    COMMAND uname -m
+    COMMAND tr -d '\n'
+    OUTPUT_VARIABLE ARCHITECTURE
+)
+
+if("${ARCHITECTURE}" STREQUAL "x86_64")
+    message("Building JSON for x86_64")
+elseif("${ARCHITECTURE}" STREQUAL "aarch64")
+    message("Building JSON for aarch64")
+else()
+    message(FATAL_ERROR "Unsupported architecture. JSON is only supported on x86_64 and aarch64.")
```

Comment:
Yes, printing out `${ARCHITECTURE}` will be helpful.

Reply:
> looks like this is all about reusing the tests at https://github.com/valkey-io/valkey-bloom/tree/unstable/tests? or there is more?

Yes, it's for reusing the pytest infrastructure in valkey-bloom.

Reply:
hmm this dependency is a bit odd - does it make sense to pull the pytest infra into the Valkey repo instead? We are discussing migrating to python based integration test infra for Valkey too.

Reply:
Yes, I would suggest we should pull it out in the valkey repo instead. However, we can change the line later when we have it separately.

Reply:
Why not move the testing framework to it's own repo to start with? It only needs to be pulled in for running the tests, not when just building the module.

I'd rather see people spend time on a README for the test framework than on an RFC. Or we can do both in parallel...?

Reply:
> I'd rather see people spend time on a README for the test framework than on an RFC. Or we can do both in parallel...?

I just want us to have some long term plan for testing and not just build it up organically. The reason is that Amazon internally has a similar testing framework that is extremely complex because nobody really maintains it and just bolt on random features. There are many cases where it's possible to do the same way 2 or 3 different ways. I think the same is true of our TCL framework, just to a smaller scale. 

So whether it be an RFC or just some direction in a new repo, I don't care which.